joblib/numpy_pickle.py

```diff
@@ -329,44 +338,49 @@ def load_build(self):
         dispatch[pickle.BUILD] = load_build
 
 
-class ZipNumpyUnpickler(NumpyUnpickler):
-    """A subclass of our Unpickler to unpickle on the fly from
-    compressed storage."""
-
-    def __init__(self, filename, file_handle):
-        NumpyUnpickler.__init__(self, filename,
-                                file_handle,
-                                mmap_mode=None)
-
-    def _open_pickle(self, file_handle):
-        return BytesIO(read_zfile(file_handle))
+def _dump(value, f, protocol):
+    """Helper function that wraps the call to joblib internal pickler."""
+    pickler = NumpyPickler(f, protocol=protocol)
+    pickler.dump(value)
 
 
 ###############################################################################
 # Utility functions
 
-def dump(value, filename, compress=0, cache_size=100, protocol=None):
-    """Fast persistence of an arbitrary Python object into one or multiple
-    files, with dedicated storage for numpy arrays.
+def dump(value, filename, compress=0, protocol=None, compress_method='zlib',
+         cache_size=None):
+    """Persist an arbitrary Python object.
+
+    Fast persistence of an arbitrary Python object into one file with
+    dedicated storage for numpy arrays.
```

Comment:
Not sure what "dedicated storage for numpy arrays" meant, maybe it is related to .npy auxiliary files, in which case, I would just remove this and use "Persist an arbitrary object into one file" as the first line of the docstring.